### PR TITLE
Enable compact list notation in config files

### PR DESCRIPTION
**Pull request by Neverlord**
Adds support for `from..to..step` syntax (borrowed form `bash` & friends) in config files (steps are optional). Particularly useful for defining affinity groups (see https://github.com/actor-framework/actor-framework/pull/980).

Two motivating examples:
- `[1..3]` expands to `[1, 2, 3]`
- `[4..-4..-2]` expands to `[4, 2, 0,- 2, -4]`

### CODE REVIEW

File: libcaf_core/test/detail/parser/read_number.cpp [hunk 1]
```diff
@@ -269,4 +301,44 @@ CAF_TEST(fractional mantissa with negative exponent) {
...
+CAF_TEST(ranges can use negative step values) {
+  CHECK_RANGE("6..2..-2", 6, 4, 2);
+  CHECK_RANGE("8..3..-3", 8, 5);
+}
+
+CAF_TEST(ranges can use signed integers) {
+  CHECK_RANGE("+2..+6..+2", 2, 4, 6);
+  CHECK_RANGE("+6..+2..-2", 6, 4, 2);
+  CHECK_RANGE("+2..-2..-2", 2, 0, -2);
+  CHECK_RANGE("-2..+2..+2", -2, 0, 2);
```
Comment (josephnoir) [inline libcaf_core/test/detail/parser/read_number.cpp]: What happens for a range like this: `"-2..+2..-2"`?

File: libcaf_core/caf/detail/parser/read_ini.hpp [hunk 2]
```diff
@@ -183,16 +184,17 @@ void read_ini_uri(State& ps, Consumer&& consumer) {
   // clang-format on
 }
 
-template <class State, class Consumer>
-void read_ini_value(State& ps, Consumer&& consumer) {
+template <class State, class Consumer, class InsideList>
+void read_ini_value(State& ps, Consumer&& consumer, InsideList inside_list) {
```
Comment (josephnoir) [inline libcaf_core/caf/detail/parser/read_ini.hpp]: The type `InsideList` is passed to `read_number_or_timespan` as the type for `EnableRange`, does the tag have a different meaning in both cases?

Comment (Neverlord) [reply libcaf_core/caf/detail/parser/read_ini.hpp]: That' correct. We enable the range syntax when inside a list. 🙂

File: libcaf_core/test/detail/parser/read_number.cpp (hunk 1 above)
Comment (Neverlord) [reply libcaf_core/test/detail/parser/read_number.cpp]: You'll get an error. I've added a new test case for that.

**Pull request merged**